src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -502,9 +499,10 @@ SHAMap::onlyBelow (SHAMapTreeNode* node) const
 
     // An inner node must have at least one leaf
     // below it, unless it's the root_
-    assert (node->hasItem () || (node == root_.get ()));
+    auto leaf = static_cast<SHAMapTreeNode*>(node);
+    assert (leaf->hasItem () || (leaf == root_.get ()));
```

Comment:
Yeah, it's interesting. In principle, it looks like it could, but in practice, it looks like it never is, at least as currently implemented.

I audited this by looking for all calls to `root_->isLeaf()` and `root_->isInner()`, putting breakpoints on the "leaf" option, and running. It never hit those breakpoints. That's obviously not conclusive, but it points in that direction. `root_` is also created as an inner node, both in the old code and new.

I guess the places that check `root_` for being a leaf are either artifacts of an older (more complicated?) implementation, or a future-proofing double-check.